test/test_bin.c

```diff
@@ -371,3 +440,17 @@ void test_bin_patch_fetch_bytes(void)
 
     TEST_ASSERT_BIN_REQ_EXP_BIN(req_get, sizeof(req_get), resp_get_expected, sizeof(resp_get_expected));
 }
+
+void test_bin_export(void)
+{
+    const char resp_expected[] =
+        "A4 "                       // map with 4 elements
+        "18 1A 1A 00 BC 61 4E "     // int 12345678
+        "18 71 FA 41 61 99 9a "     // float 14.10
+        "18 72 FA 40 a4 28 f6 "     // float 5.13
+        "18 73 16 ";                // int 22
+
```

Comment:
It looks like the test relies on some values set by another test run before. This should be avoided as it might lead to problems in test maintenance. If this is really needed squash the tests to a single test function otherwise preset the values in the test. I think this is a general pattern in the tests.

Reply:
You are right, the tests should be independent of each other and not depend on the order they are called, as it is the case at the moment. I opened an issue to go through all tests and fix this, so I'll keep it for this PR.